shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1321,14 +1334,21 @@ public void writeFields(DataOutput out) throws IOException {
             out.writeUTF(target);
             out.writeUTF(source);
             out.writeUTF(delegationToken == null ? "" : delegationToken);
+            for (ConditionalAttributeUpdate entry : attributeUpdates) {
```

Comment:
You need to write the number of entries. I see you use a while loop to read them, but you are essentially tying your hands now as you won't be able to add any new fields in the future.

Encoding the number of attribute updates enables you to do a for loop in the read method, and then if (when) we add new fields at the end, we do not have to go out of our way to make it happen.

Reply:
Good catch, thanks. I have done that writing and reading first the number of entries.